### PR TITLE
HPCC-27138 Sync Ubuntu / OSX / Windows vcpkg usage

### PR DESCRIPTION
**Pull request by GordonSmith**
## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [x] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [x] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by GordonSmith** [comment]
Depends on:  https://github.com/hpcc-systems/HPCC-Platform/pull/15739

**Comment by GordonSmith** [comment]
@richardkchapman @ghalliday rebased and ready to be pulled (assuming the builds pass)

**Comment by GordonSmith** [comment]
@richardkchapman this is ready to be pulled - the Build with vcpkg (macos-10.15) was replaced with Build with vcpkg (macos-11)

File: .github/workflows/build-vcpkg.yml
Comment (richardkchapman) [inline .github/workflows/build-vcpkg.yml:85]: This will require me to update the "required tests" setting again (note to self to remember to do so after merging this).

**Review by richardkchapman — APPROVED** [review]

File: .github/workflows/build-vcpkg.yml
Comment (GordonSmith) [reply .github/workflows/build-vcpkg.yml:85]: Sorry about that - but I thought 10.15 was Big Sur until I double checked...

**Review by ghalliday — APPROVED** [review]

**Comment by ghalliday** [comment]
@richardkchapman I'll let you merge this so you can synchronize your other change

**Pull request merged**
